packages/cinterop/src/jvm/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -157,9 +157,17 @@ actual object RealmInterop {
         realmc.realm_config_set_data_initialization_function(config.cptr(), callback)
     }
 
-    actual fun realm_open(config: RealmConfigurationPointer, dispatcher: CoroutineDispatcher?): LiveRealmPointer {
-        // create a custom Scheduler for JVM if a Coroutine Dispatcher is provided other wise pass null to use the generic one
+    actual fun realm_open(config: RealmConfigurationPointer, dispatcher: CoroutineDispatcher?): Pair<LiveRealmPointer, Boolean> {
+        // Configure callback to track if the file was created as part of opening
+        var fileCreated = false
```

Comment:
Should access to this be synchronized? I would have expected all of this to happen on the same thread, but from the Darwin implementation it looks like it isn't 🤔

Reply:
It is happening on the same thread. but on Native Kotlin had a compile check around capturing lambdas + I needed to freeze the callback on Native so the GC would work.